internal/database/repository/node.go

```diff
@@ -40,56 +59,231 @@ func applyNodeFilters(builder *sql.Builder, filters NodeFilters) {
+	}
+	return &node, nil
+}
+
+func (s *PostgresRepository) DeleteNodesNotInIDs(ctx context.Context, ids []string, deletedAtNano int64) error {
+	const q = `
+UPDATE nodes 
+SET deleted_at_nano = @deleted_at_nano
+WHERE deleted_at_nano IS NULL AND NOT (id = ANY(@ids))`
+	_, err := s.dbpool.Exec(
+		ctx,
+		q,
+		pgx.NamedArgs{
+			"deleted_at_nano": deletedAtNano,
+			"ids":             ids,
+		},
+	)
+	return err
+}
+
+func (s *PostgresRepository) InsertNodeUtilizations(
+	ctx context.Context,
+	nus []*dao.PartitionNodesUtilDAOInfo,
+) error {
+	insertSQL := `INSERT INTO partition_nodes_util (id, cluster_id, partition, nodes_util_list)
+		VALUES (@id, @cluster_id, @partition, @nodes_util_list)`
+
+	for _, nu := range nus {
+		_, err := s.dbpool.Exec(ctx, insertSQL,
 			pgx.NamedArgs{
-				"id":           ulid.Make().String(),
-				"node_id":      n.NodeID,
-				"partition":    partition,
-				"host_name":    n.HostName,
-				"rack_name":    n.RackName,
-				"attributes":   n.Attributes,
-				"capacity":     n.Capacity,
-				"allocated":    n.Allocated,
-				"occupied":     n.Occupied,
-				"available":    n.Available,
-				"utilized":     n.Utilized,
-				"allocations":  n.Allocations,
-				"schedulable":  n.Schedulable,
-				"is_reserved":  n.IsReserved,
-				"reservations": n.Reservations,
+				"id":              ulid.Make().String(),
+				"cluster_id":      nu.ClusterID,
+				"partition":       nu.Partition,
+				"nodes_util_list": nu.NodesUtilList,
 			})
 		if err != nil {
-			return fmt.Errorf("could not insert application into DB: %v", err)
+			return fmt.Errorf("could not insert node utilizations into DB: %v", err)
 		}
+
 	}
 	return nil
 }
 
-func (s *PostgresRepository) GetNodesPerPartition(ctx context.Context, partition string, filters NodeFilters) ([]*dao.NodeDAOInfo, error) {
+func (s *PostgresRepository) GetNodeUtilizations(
+	ctx context.Context,
+	filters NodeUtilFilters,
+) ([]*dao.PartitionNodesUtilDAOInfo, error) {
+	queryBuilder := sql.NewBuilder().
+		SelectAll("partition_nodes_util", "").
+		OrderBy("id", sql.OrderByDescending)
+
+	applyNodeUtilFilters(queryBuilder, filters)
+
+	var nodesUtil []*dao.PartitionNodesUtilDAOInfo
+
+	query := queryBuilder.Query()
+	args := queryBuilder.Args()
+	rows, err := s.dbpool.Query(ctx, query, args...)
+	if err != nil {
+		return nil, fmt.Errorf("could not get node utilizations from DB: %v", err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var nu dao.PartitionNodesUtilDAOInfo
+		var id string
+		err := rows.Scan(&id, &nu.ClusterID, &nu.Partition, &nu.NodesUtilList)
+		if err != nil {
+			return nil, fmt.Errorf("could not scan node utilizations from DB: %v", err)
+		}
+		nodesUtil = append(nodesUtil, &nu)
+	}
+	return nodesUtil, nil
+}
+
+func (s *PostgresRepository) GetNodesPerPartition(ctx context.Context, partition string, filters NodeFilters) ([]*model.Node, error) {
 	queryBuilder := sql.NewBuilder().
 		SelectAll("nodes", "").
```

Comment:
Not in scope of this PR, but I'm not a huge fan of scanning based on position. We should probably change the interface to list the fields. We shouldn't look into the db field order to make the scan correct. 

WDYT @richscott @sudiptob2

Reply:
Good Idea, I think depending on the order is error-prone.